Duplicati/Library/Common/IO/SystemIOWindows.cs

```diff
@@ -29,20 +29,47 @@ namespace Duplicati.Library.Common.IO
 {
     public struct SystemIOWindows : ISystemIO
     {
-        private const string UNCPREFIX = @"\\?\";
-        private const string UNCPREFIX_SERVER = @"\\?\UNC\";
-        private const string PATHPREFIX_SERVER = @"\\";
+        private const string UNCPREFIX = PathInternalWindows.ExtendedDevicePathPrefix;
+        private const string UNCPREFIX_SERVER = PathInternalWindows.UncExtendedPathPrefix;
+        private const string PATHPREFIX_SERVER = PathInternalWindows.UncPathPrefix;
+        private const string PATHPREFIX_SERVER_ALT = @"//";
         private static readonly string DIRSEP = Util.DirectorySeparatorString;
 
+        /// <summary>
+        /// Prefix path with one of the UNC prefixes, but only if it's a fully
+        /// qualified path with no relative components (i.e., with no "." or
+        /// ".." as part of the path).
+        /// </summary>
         public static string PrefixWithUNC(string path)
         {
             if (IsPrefixedWithUNC(path))
             {
+                // For example: \\?\C:\Temp\foo.txt or \\?\UNC\example.com\share\foo.txt
+                return path;
+            }
+            else if (IsPrefixedWithBasicUNC(path) && !HasRelativePathComponents(path))

```

Comment:
Just to clarify, does this mean that UNC paths with relative path components will not get the `\\?\UNC\` prefix?

Reply:
Yes, I think that's right.  As long as there are what I've called "relative components" in the path, the normalization step is needed to resolve them, so we can't prefix with `\\?\`, etc.

That does mean, though, that if someone, say through one of the command line programs, happens to provide what I was calling a "problematic path" but with a relative component, we won't be prefixing it and Windows will "normalize" the path.  A path like this, `\\example.com\share\problem.\folder\..\file.txt.` with `problem.` being the problematic part of the path, will get modified by the likes of Path.GetFullPath to `\\example.com\share\problem\file.txt` where `problem.` got converted to just `problem` without the dot.  If someone needs to manually provide a problematic path through one of the command line programs, it would need to be devoid of these relative components.  So, in this case, someone could pass in `\\example.com\share\problem.\file.txt` instead and the `problem.` with the dot would be preserved.